src/storage/page.rs

```diff
@@ -73,3 +90,126 @@ impl<const LEN: usize> AsRef<[u32; LEN]> for Page<LEN> {
         &self.data
     }
 }
+
+impl<const LEN: usize> CustomSerialize for Pagepool<LEN> {
+    fn serialize(
+        &self,
+        bufmans: std::sync::Arc<crate::models::buffered_io::BufferManagerFactory>,
+        version: crate::models::versioning::Hash,
+        cursor: u64,
+    ) -> Result<u32, crate::models::buffered_io::BufIoError> {
+        let bufman = bufmans.get(&version)?;
+        let start_offset = bufman.cursor_position(cursor)? as u32;
+        let total_items = self.inner.len();
+
+        // Write total length in the memory
+        bufman.write_u32_with_cursor(cursor, total_items as u32)?;
+
+        for (pos, item) in self.inner.iter().enumerate() {
+            Page::<LEN>::serialize(item, bufmans.clone(), version, cursor)?;
+
+            if pos == total_items - 1 {
+                bufman.write_u32_with_cursor(cursor, u32::MAX)?;
+            } else {
+                let current = bufman.cursor_position(cursor)?;
+                bufman.write_u32_with_cursor(cursor, current as u32)?;
+            }
+        }
+
+        Ok(start_offset)
+    }
+
+    fn deserialize(
+        bufmans: std::sync::Arc<crate::models::buffered_io::BufferManagerFactory>,
+        file_index: crate::models::lazy_load::FileIndex,
+        cache: std::sync::Arc<crate::models::cache_loader::NodeRegistry>,
+        max_loads: u16,
+        skipm: &mut std::collections::HashSet<u64>,
+    ) -> Result<Self, crate::models::buffered_io::BufIoError> {
+        match file_index {
+            crate::models::lazy_load::FileIndex::Valid {
+                offset: FileOffset(offset),
+                version_id,
+                ..
+            } => {
+                let bufman = bufmans.get(&version_id)?;
+                let cursor = bufman.open_cursor()?;
+
+                bufman.seek_with_cursor(cursor, SeekFrom::Start(offset as u64))?;
+
+                let total_length = bufman.read_u32_with_cursor(cursor)? as usize;
+
+                let mut page_pool = Pagepool::<LEN>::default();
+                for _ in 0..total_length {
+                    let page = Page::<LEN>::deserialize(
+                        bufmans.clone(),
+                        file_index.clone(),
```

Comment:
instead of using this `file_index`, you should create a new `FileIndex` like this:
```rust
let current_offset = bufman.cursor_position(cursor)?;
let page_file_index = FileIndex {
    offset: FileOffset(current_offset as u32),
    version_id,
    version_number,
};

Reply:
Ah my bad